runtime/libc/libc_backing.c

```diff
@@ -516,7 +516,7 @@ u32 wasm_geteuid() {
 
 #define SYS_GET_TIME 228
 struct wasm_time_spec {
-    u32 sec;
+    u64 sec;
     u32 nanosec;
 };
```

Comment:
Can you cite where in musl this is set as 64 bit?

Reply:
In musl: https://git.musl-libc.org/cgit/musl/tree/arch/i386/bits/alltypes.h.in?id=38143339646a4ccce8afe298c34467767c899f51#n33

Reply:
We use x86_64 as our base though. (This was obviously dumb, but it is what it is.) I guess it might be 64bits anyway? https://github.com/Others/musl/blob/master/arch/x86_64/bits/alltypes.h.in

Reply:
u32 is likely `unsigned int` or `unsigned long`, and u64 is `unsigned long long` if I remember correctly. And the alltypes.h in x86_64 that you linked to also defines `time_t` as `long long`. 
Also to be clear, clock_gettime returns gibberish without this fix!

Reply:
Just noticed this: https://github.com/gwsystems/wasmception-musl/commit/d92987dfced63df9a0e8268b81ab7d9dee4312ae#diff-1f02b3cd6244cd1511b2020556d1c584R22

This diff suggests time_t was originally long in x86_64 but you seemed to have changed it to long long. That explains the bug, because that change wasn't propagated to the libc in awsm. I think you know best to fix this the right way, just sharing for more information!